app/controllers/works_controller.rb

```diff
@@ -0,0 +1,86 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :edit, :update, :destroy]
+  before_action :require_login, only: [:create, :update, :destroy, :upvote]
+
+  def index
+    @works = Work.all
+  end
+
+  def show
+    if @work.nil?
+      redirect_to work_path
+      return
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(works_param)
+
+    if @work.save
+      flash[:success] = "Your #{@work.category} was added successfully!"
+      redirect_to work_path(@work.id)
+      return
+    else
+      flash.now[:error] = "Oops! Media already exists."
```

Comment:
I love the idea of having more specific messages for specific situations, but surfacing validation messages is a more fool-proof way to do that.
The way this is currently written, it assumes that if something goes wrong, it's because the work already exists when there are other things that can go wrong.

Reply:
In fact, when I try to add a duplicate work, I get the following validation message anyway
`title has already been taken`